### PR TITLE
Feature/contribution totals

### PR DESCRIPTION
**Pull request by yachtcaptain23**
Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Tagged reviewers.
- [ ] Integrated piwik/matomo (for code that adds new buttons).
- [ ] Addressed or ignored all brakeman warnings

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

File: app/jobs/cache/piwik_data_job.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  def perform
+    site = Piwik::Site.load(6)
+    Rails.cache.write(SEO_INFO, site.seo_info.data)
+    Rails.cache.write(PIWIK_VISITS_SUMMARY, site.visits.getVisits(:idSite => 6, :period => :month, :date => :last12).result.to_json)
+    Rails.cache.write(PIWIK_EVENTS_CATEGORY, site.events.getCategory(idSite: 6, period: 'week', date: 1.week.ago.strftime("%Y-%m-%d")).data.to_json)
+    Rails.cache.write(PIWIK_DEVICES_DETECTION_TYPE, site.devices_detection.getType(idSite: 6, period: 'week', date: 1.week.ago.strftime("%Y-%m-%d")).data.to_json)
+    Rails.cache.write(PIWIK_DEVICES_DETECTION_BROWSER_VERSIONS, site.devices_detection.getBrowserVersions(idSite: 6, period: 'week', date: 1.week.ago.strftime("%Y-%m-%d")).data.to_json)
+    Rails.cache.write(PIWIK_CACHE_LAST_UPDATED, Time.now)
+  end
```
Comment (nvonpentz) [inline app/jobs/cache/piwik_data_job.rb]: If the correct piwik keys are not available, then what is displayed on the dashboard?  Is it possible to have sample data displayed in the case that the keys do not exist?

Comment (yachtcaptain23) [reply app/jobs/cache/piwik_data_job.rb]: If the correct piwik keys are not available, then it should just render the previous data.

File: config/sidekiq.yml [hunk 2]
```diff
@@ -39,6 +39,14 @@
     cron: "0 */2 * * *"
     description: "Refreshes the redis cache for browser channel json"
     queue: transactional
+  Cache::PiwikDataJob:
+    cron: "45 2 * * *"
+    description: "Refreshes the redis cache for piwik information"
```
[the remark this answers is not in the document]
Comment (yachtcaptain23) [reply config/sidekiq.yml]: Sure thing

File: app/services/eyeshade/base_api_client.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Eyeshade::BaseApiClient < BaseApiClient
```
[the remark this answers is not in the document]
Comment (yachtcaptain23) [reply app/services/eyeshade/base_api_client.rb]: Was planning on doing so after this PR. I'm curious to see the data first.

[the remark this answers is not in the document]
Comment (nvonpentz) [reply app/services/eyeshade/base_api_client.rb]: Ok maybe we should create an issue for this

**Comment by yachtcaptain23** [comment]
@nvonpentz I disagree that we rename to Matomo. Based on the number of references with either `grep` or `ag`, our codebase consistently references `piwik`. The gem itself `autometal-piwik`, as part of the `matomo-org`.

**Pull request merged**